scripts/brands-page/brands.js

```diff
@@ -0,0 +1,291 @@
+const debouncedFetchAndRenderBrands = debounce(fetchAndRenderBrands, 300)
+
+/**
+ * Handles sorting of the brands table based on the specified column.
+ *
+ * Toggles the sorting order (ascending or descending) if the same column is clicked consecutively.
+ * Updates the current sort column and fetches the updated list of brands.
+ *
+ * @param {string} column - The name of the column to sort by (e.g., 'id' or 'name').
+ */
+
+
+export const handleSorting = (column) => {
+    // Toggle sort order if the same column is clicked
+    if (currentSortColumn === column) {
+        currentSortOrder = currentSortOrder === 'ASCENDING' ? 'DESCENDING' : 'ASCENDING'
+    } else {
+        // Set to ASCENDING for a new column
+        currentSortColumn = column
+        currentSortOrder = 'ASCENDING'
+    }
+
+    console.log(`Sorting by ${currentSortColumn} in ${currentSortOrder} order`)
+
+    // Fetch data with the updated sort order
+    fetchAndRenderBrands()
+}
+
+/**
+ * Handles filtering of the brands table based on user input.
+ *
+ * Applies a "LIKE" filter on the 'name' field if a value is provided in the input.
+ * Clears the filters if the input is empty. Resets the current page to the first page and fetches the updated list.
+ *
+ * @param {Event} e - The input event triggered by the filter input element.
+ */
+export let handleFiltering = (e) => {
+    const filterValue = e.target.value.trim()
+    const filterField = e.target.dataset.field // Use a data attribute to identify the field
+
+    if (filterValue && filterField) {
+        filters = [
+            {
+                field: filterField, // The field to filter (either "id" or "name")
+                type: filterField === 'id' ? 'EQUAL' : 'LIKE', // Use "EQUAL" for id, "LIKE" for name
+                value: [filterValue]
+            }
+        ]
+    } else {
+        filters = [] // Clear filters if input is empty
+    }
+
+    currentPage = 1 // Reset to the first page
+    fetchAndRenderBrands()
+}
+
+/**
+ * Renders an error message into the HTML element with the ID of 'brandsList'.
+ *
+ * The error message indicates that the brands could not be loaded, and it is styled using Bootstrap classes.
+ */
+
+
+const renderError = () => {
+    const listContainer = document.getElementById('brandsList')
+
+
+    if (listContainer) {
+        listContainer.innerHTML = `
+        <li class="list-group-item text-danger">
+            Failed to load brands. Please try again later.
+        </li>
+    `
+    }
+}
+
+// Initialize fetching and rendering on page load
+document.addEventListener('DOMContentLoaded', fetchAndRenderBrands)
+
+document.addEventListener('DOMContentLoaded', () => {
+    const idHeader = document.querySelector('th:nth-child(2)')
+    const nameHeader = document.querySelector('th:nth-child(3)')
+
+    if (idHeader && !idHeader.dataset.listenerAdded) {
+        idHeader.addEventListener('click', () => handleSorting('id'))
+        idHeader.dataset.listenerAdded = true
```

Comment:
Assigned expression type boolean is not assignable to type string.
I know it will work anyway, just a warning :)